### ISSUE TITLE
Not able to read model.json file of Common Data Service

### ISSUE THREAD

**Issue by bparchuri** [Azure/spark-cdm-connector#35]
We are trying to read to model.json(Common Data Service) using the connector -> https://github.com/Azure/spark-cdm-connector/blob/master/documentation/overview.md in Azure Data Bricks and installed the jar file 14.0 in cluster as shown in the first screenshot, later after running the command we are getting the error in the notebook as shown in the second screenshot. Are we missing anything? Is it going to support model.json or only manifestmodel.json?

![image](https://user-images.githubusercontent.com/58940818/85151736-6fbee580-b254-11ea-9694-f20f1a8c4b17.png)
![image](https://user-images.githubusercontent.com/58940818/85151787-806f5b80-b254-11ea-8c4e-247958907d65.png)

java.lang.RuntimeException: Manifest doesn&apos;t exist: model.json
---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
<command-3329376230467366> in <module>()
     12 .option("appId", appid)
     13 .option("appKey", appkey)
---> 14 .option("tenantId", tenantid)
     15 .load())

/databricks/spark/python/pyspark/sql/readwriter.py in load(self, path, format, schema, **options)
    170             return self._df(self._jreader.load(self._spark._sc._jvm.PythonUtils.toSeq(path)))
    171         else:
--> 172             return self._df(self._jreader.load())
    173 
    174     @since(1.4)

/databricks/spark/python/lib/py4j-0.10.7-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1255         answer = self.gateway_client.send_command(command)
   1256         return_value = get_return_value(
-> 1257             answer, self.gateway_client, self.target_id, self.name)
   1258 
   1259         for temp_arg in temp_args:

Regards,
Bhargav Parchuri

**Comment by sricheta92** [comment]
Spark-CDM connector supports reading manifest.cdm.json as well as model.json. In your case, I think your manifest starts with http://storage_account_name/container/{manifest path}, which is not correct. Please read : https://github.com/Azure/spark-cdm-connector/blob/master/documentation/overview.md#common-options

The {manifest path} is what you should put in 
.option(“manifest”, {manifest path})

**Comment by bparchuri** [comment]
Ok! Thank you for the prompt response!

As per your suggestion I have changed Manifest Path and able to read model.json. This is the exactly what we are looking again!

Great tool!

Have a nice weekend!

Regards,
[cid:image001.png@01D6467A.7EAED5F0]



Bhargav Parchuri
Senior Consultant - Digital Expertise

[mobile] +32 492 145995

**Comment by rahuljim** [comment]
we are still getting same error please help
![image](https://user-images.githubusercontent.com/58183111/85251298-99575700-b476-11ea-9048-0ffc82ea1a97.png)

this is our directory and we are getting below error :
spark.read.format("com.microsoft.cdm") .option("storage", ".dfs.core.windows.net") .option("container", "cdmtest") .option("manifest", "/model.json") .option("entity", "contact") .option("appId",-ba011e") .option("appKey", "7Q3") .option("tenantId", "3503d") .load()

/databricks/spark/python/pyspark/sql/readwriter.py in load(self, path, format, schema, **options)
170 return self._df(self._jreader.load(self._spark._sc._jvm.PythonUtils.toSeq(path)))
171 else:
--> 172 return self._df(self._jreader.load())
173
174 @SInCE(1.4)

/databricks/spark/python/lib/py4j-0.10.7-src.zip/py4j/java_gateway.py in call(self, *args)
1255 answer = self.gateway_client.send_command(command)
1256 return_value = get_return_value(
-> 1257 answer, self.gateway_client, self.target_id, self.name)
1258
1259 for temp_arg in temp_args:

/databricks/spark/python/pyspark/sql/utils.py in deco(*a, **kw)
61 def deco(*a, **kw):
62 try:
---> 63 return f(*a, **kw)
64 except py4j.protocol.Py4JJavaError as e:
65 s = e.java_exception.toString()

/databricks/spark/python/lib/py4j-0.10.7-src.zip/py4j/protocol.py in get_return_value(answer, gateway_client, target_id, name)
326 raise Py4JJavaError(
327 "An error occurred while calling {0}{1}{2}.\n".
--> 328 format(target_id, ".", name), value)
329 else:
330 raise Py4JError(

Py4JJavaError: An error occurred while calling o498.load.
: java.lang.RuntimeException: Manifest doesn't exist: model.json
at com.microsoft.cdm.utils.CDMModelCommon.getEntityDec(CDMModelCommon.scala:82)
at com.microsoft.cdm.read.CDMDataSourceReader.(CDMDataSourceReader.scala:28)
at com.microsoft.cdm.DefaultSource.createReader(DefaultSource.scala:99)
at com.microsoft.cdm.DefaultSource.createReader(DefaultSource.scala:19)
at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$SourceHelpers.createReader(DataSourceV2Relation.scala:155)
at org.apache.spark.sql.execution.datasources.v2.DataSourceV2Relation$.create(DataSourceV2Relation.scala:172)
at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:290)
at org.apache.spark.sql.DataFrameReader.load(DataFrameReader.scala:203)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:380)
at py4j.Gateway.invoke(Gateway.java:295)
at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
at py4j.commands.CallCommand.execute(CallCommand.java:7

**Comment by rahuljim** [comment]
i have not got resolution why it was closed

**Comment by bparchuri** [comment]
Hi Team,

Good Day!

I am facing strange issue, after reading the Data Frame, initially, I was able to Display the Data Frame and able to create table in Hive using this Data Frame.
But today, I am getting the below issue while trying to display the Data Frame which was trying to read Incident Entity from “Model.JSON”. Any clue why this is getting the below issue.


NoteBook Python Script:

readDf =( spark.read.format("com.microsoft.cdm")
.option("storage", storageAccountName)
.option("container", container)
.option("manifest", "model.json")
.option("entity", "incident")
.option("appId", appid)
.option("appKey", appkey)
.option("tenantId", tenantid)
.load())
display(readDf)

Error:

[cid:image001.jpg@01D64EDD.1A1856F0]

[cid:image002.png@01D64EDD.1A1856F0]



Bhargav Parchuri
Senior Consultant - Digital Expertise

[mobile] +32 492 145995


From: Bhargav Parchuri
Sent: Friday, June 19, 2020 8:45 PM
To: Azure/spark-cdm-connector <reply@reply.github.com>; Azure/spark-cdm-connector <spark-cdm-connector@noreply.github.com>
Cc: Author <author@noreply.github.com>
Subject: RE: [Azure/spark-cdm-connector] Not able to read model.json file of Common Data Service (#35)

Ok! Thank you for the prompt response!

As per your suggestion I have changed Manifest Path and able to read model.json. This is the exactly what we are looking again!

Great tool!

Have a nice weekend!

Regards,
[cid:image006.png@01D64EDD.1A1856F0]



Bhargav Parchuri
Senior Consultant - Digital Expertise

[mobile] +32 492 145995
